packages/ecs/src/ComponentFunctions.ts

```diff
@@ -461,9 +461,7 @@ export const getOptionalMutableComponent = <C extends Component>(
   entity: Entity,
   component: C
 ): State<ComponentType<C>> | undefined => {
-  return !bitECS.hasComponent(HyperFlux.store, entity, component)
-    ? undefined
-    : (component.stateMap[entity]! as State<ComponentType<C>> | undefined)
+  return component?.stateMap?.[entity]
```

Comment:
So we previously had something here different from getOptionalComponent where we are forcing it to be a value. This is causing tests to fail. Might be causing something at runtime to fail as well. Thoughts? It doesn't match with the return type we had to begin with.